src/repository.ts

```diff
@@ -370,9 +373,9 @@ export class LernaRepo extends Repository {
     if (!isEmpty(nextVersions)) {
       for (const pkgPath of pkgPaths) {
         const pkg = await Package.create(pkgPath);
-        const shouldBePublished = await this.isReleasable(pkg, true);
+        const shouldBePublihsed = this.shouldBePublished || (await this.isReleasable(pkg, true));
```

Comment:
The packages are still getting filtered out because `nextVersions` is empty on line 373.

Feel free to implement a fix for this however you want but I'm now thinking that we should undo adding `shouldBePublished` and instead do the following:

Add public method to `LernaRepo` called `getPackages`. The implementation may look something like this:

```typescript
  public async getPackages(): Promise<Package[]> {
    const pkgPaths = await this.getPackagePaths();
    const packages: Package[] = [];
    for (const pkgPath of pkgPaths) {
      packages.push(await Package.create(pkgPath));
    }
    return packages;
  }
```

And then the `getPackages` method in `update.ts` will look like this:
```typescript
private async getPackages(): Promise<Package[]> {
    return this.repo instanceof LernaRepo ? await this.repo.getPackages() : [this.repo.package];
  }
```

That just an idea though - you don't have to fix it that way.

Reply:
Yes that method works for getting the packages, but if we fail to set the value of `nextVersions` then in the release we wont be able to return the next release version correctly

```
return lernaRepo.packages.map((pkg) => {
      return { name: pkg.name, version: pkg.getNextVersion() };
    });
```